src/Chapter2.hs

```diff
@@ -501,7 +503,11 @@ True
 >>> isThird42 [42, 42, 0, 42]
 False
 -}
-isThird42 = error "isThird42: Not implemented!"
+isThird42 :: [Int] -> Bool
+isThird42 [] = False
+isThird42 [_] = False
+isThird42 [_, _] = False
+isThird42 (_:_:z:_) = z == 42
```

Comment:
Absolutely correct!

If you change the order of patterns it would be enough to have only two – the one with 42 and everything else. 

Note also, that you don't need to compare `z` with 42 separately, you can pattern match on `42` directly in the pattern.